metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MetricsAggregatorAthenaClient.java

```diff
@@ -0,0 +1,270 @@
+                    }
+                }
+            }
+        }
+
+        return queryResultRows;
+    }
+
+    /**
+     * Calculate aggregated metrics for all platforms in the S3 directory.
+     * @param s3DirectoryInfo
+     * @return
+     */
+    public Map<String, Metrics> getAggregatedMetricsForPlatforms(S3DirectoryInfo s3DirectoryInfo) {
+        LOG.info("Aggregating metrics for directory: {}", s3DirectoryInfo.versionS3KeyPrefix());
+        Map<String, Metrics> platformToMetrics = new HashMap<>();
+        // Calculate metrics for runexecutions
+        // TODO: Calculate metrics for taskexecutions and validationexecutions
+        AthenaTablePartition athenaTablePartition = s3DirectoryInfo.athenaTablePartition();
+        try {
+            LOG.info("Aggregating workflow executions");
+            String query = executionStatusAggregator.createRunExecutionsQuery(tableName, athenaTablePartition);
+            Map<String, ExecutionStatusMetric> platformToExecutionStatusMetric = executionStatusAggregator.createMetricByPlatform(
+                    executeQuery(query));
+            platformToExecutionStatusMetric.forEach((platform, executionStatusMetric) -> {
+                if (platformToMetrics.containsKey(platform)) {
+                    platformToMetrics.get(platform).executionStatusCount(executionStatusMetric);
+                } else {
+                    platformToMetrics.put(platform, new Metrics().executionStatusCount(executionStatusMetric));
+                }
+                LOG.info("Aggregated metrics for tool ID {}, version {}, platform {} from directory {}", s3DirectoryInfo.toolId(),
+                        s3DirectoryInfo.versionId(), platform, s3DirectoryInfo.versionS3KeyPrefix());
+            });
+        } catch (Exception e) {
+            LOG.error("Could not aggregate metrics for tool ID {}, version {}", s3DirectoryInfo.toolId(), s3DirectoryInfo.versionId(), e);
+        }
+
+        numberOfDirectoriesProcessed.incrementAndGet();
+        LOG.info("Processed {} directories", numberOfDirectoriesProcessed);
+        return platformToMetrics;
+    }
+
+    /**
+     * A record that contains information about a single row of results from an Athena query execution.
+     * @param columnNameToColumnIndex Map of column names to column index to be used with the columnValues list.
+     * @param columnValues List of column values.
+     */
+    public record QueryResultRow(Map<String, Integer> columnNameToColumnIndex, List<String> columnValues) {
+        /**
+         * Get the column value that corresponds with the column name.
+         * @param columnName
+         * @return
+         */
+        public Optional<String> getColumnValue(String columnName) {
+            Integer columnIndex = columnNameToColumnIndex.get(columnName);
+            return columnIndex == null || columnValues.get(columnIndex) == null ? Optional.empty() : Optional.of(columnValues.get(columnIndex));
```

Comment:
No need to change, but you could simplify (arguably) this to:

`return columnIndex == null ? Optional.empty() : Optional.ofNullable(columnValues.get(columnIndex))`